cycli/neo4j.py

```diff
@@ -40,10 +45,102 @@ def cypher(self, query):
         return results, duration
 
     def labels(self):
-        return sorted(list(self.graph.node_labels))
+        if self.__labels is None:
+            # directly query from resource - as py2neo does caching of node_labels
+            labels_resource = Resource(self.graph.uri.string + "labels")
+            self.__labels = sorted(list(frozenset(labels_resource.get().content)))
+        return self.__labels
 
     def relationship_types(self):
-        return sorted(list(self.graph.relationship_types))
+        if self.__relationship_types is None:
+            # directly query from resource - as py2neo does caching of relationship_types
+            relationship_types_resource = Resource(self.graph.uri.string + "relationship/types")
+            self.__relationship_types = sorted(list(frozenset(relationship_types_resource.get().content)))
+        return self.__relationship_types
+
+    def constraints(self):
+        if self.__constraints is None:
+            constraints = []
+            constraints_resource = Resource(self.graph.uri.string + "schema/constraint")
+            constraints_content = list(constraints_resource.get().content)
+            for i in constraints_content:
+                constraint = dict(i)
+                constraint["property_keys"] = list(constraint["property_keys"])
+                constraints.append(constraint)
+
+            self.__constraints = constraints
+        return self.__constraints
+
+    def indexes(self):
+        if self.__indexes is None:
+            indexes = []
+            for label in self.labels():
+                index_resource = Resource(self.graph.uri.string + "schema/index/" + label)
+                indexes_content = list(index_resource.get().content)
+                for i in indexes_content:
+                    index = dict(i)
+                    index["property_keys"] = list(index["property_keys"])
+                    indexes.append(index)
+
+            self.__indexes = indexes
+        return self.__indexes
+
+    # when schema changes - user can call refresh
+    def refresh(self):
+        self.__labels = None
+        self.__relationship_types = None
+        self.__indexes = None
+        self.__constraints = None
+        self.labels()
+        self.relationship_types()
+        self.indexes()
+        self.constraints()
+
+    def print_labels(self):
+        results = self.labels()
+        print("Labels")
```

Comment:
That's because `rjust` does nothing if the string is longer than the argument.

```python
>>> "xxxx".rjust(4)
'xxxx'
>>> "xxxx".rjust(2)
'xxxx'
>>> "xxxx".rjust(6)
'  xxxx'
```